### ISSUE TITLE
make crash

### ISSUE THREAD

**Issue by MrReskill** [MCMrARM/mcpelauncher-linux#264]
after cmake i want to type make but it call to me an error: 
[ 54%] Building CXX object CMakeFiles/mcpelauncher.dir/src/main.cpp.o
In file included from /home/albanw/Téléchargements/mcpelauncher-linux/src/main.cpp:38:0:
/home/albanw/Téléchargements/mcpelauncher-linux/src/extract.h:4:17: fatal error: zip.h: Aucun fichier ou dossier de ce type
compilation terminated.
CMakeFiles/mcpelauncher.dir/build.make:446 : la recette pour la cible « CMakeFiles/mcpelauncher.dir/src/main.cpp.o » a échouée
make[2]: *** [CMakeFiles/mcpelauncher.dir/src/main.cpp.o] Erreur 1
CMakeFiles/Makefile2:67 : la recette pour la cible « CMakeFiles/mcpelauncher.dir/all » a échouée
make[1]: *** [CMakeFiles/mcpelauncher.dir/all] Erreur 2
Makefile:127 : la recette pour la cible « all » a échouée
make: *** [all] Erreur 2

(I have installed all requirements)

**Comment by coolkidsrule101** [comment]
no you havent because one if you did git clone the gapi is not include witch cause the make to fail
click on the gapi and download it and move the file to the gapi folder in the launcher ok... if any other problems tell me or mcmrarm

**Comment by coolkidsrule101** [comment]
wait what you language

**Comment by guppy42** [comment]
I don't speak french but the ["fatal error: zip.h" crops up quite often](https://github.com/MCMrARM/mcpelauncher-linux/search?q=fatal+error%3A+zip.h&type=Issues&utf8=%E2%9C%93)

The error occurs because your missing
`libzip-dev:i386`

which is listed in the [requirements](https://github.com/MCMrARM/mcpelauncher-linux/blob/master/README.md#required-packages-on-ubuntu) 

basically you need to excute;

`sudo dpkg --add-architecture i386`

`sudo apt install cmake gcc-multilib g++-multilib zlib1g-dev:i386 libx11-dev:i386 libzip-dev:i386 libpng-dev:i386 uuid-dev:i386 libcurl4-openssl-dev:i386 libssl-dev:i386 libgles2-mesa-dev:i386`

`sudo apt install libgtk2.0-0:i386 libgtkglext1:i386 libasound2:i386 libnss3:i386 libxss1:i386 libgconf2-4:i386 libxtst6:i386 libudev1:i386`

`sudo apt install protobuf-compiler libprotobuf-dev:i386`

once you've run all of those commands you have installed the requirements and are ready to move onto [compiling](https://github.com/MCMrARM/mcpelauncher-linux/blob/master/README.md#compiling)


The requirements assume your running ubuntu ( or one of the derived versions like kbuntu,lbuntu,mint, etc ) if your something else the packages names may well differ

**Comment by MrReskill** [comment]
Thanks it's work but i have a 1.2.5 MC:PE how to update on 1.2.7 ?

**Comment by guppy42** [comment]
It's in the wiki 
https://github.com/MCMrARM/mcpelauncher-linux/wiki/Updating-Minecraft#for-the-apk-download-branch
I should really update it during the holidays

In short run
`mcpelauncher setup`
To  have it update to the latest version (if you've entered the beta program it will also download betas)

**Comment by coolkidsrule101** [comment]
Bruh

**Comment by coolkidsrule101** [comment]
Really dude

**Comment by MrReskill** [comment]
i get it: 
![capture du 2017-12-21 20-01-38](https://user-images.githubusercontent.com/16138643/34270814-d825a972-e689-11e7-94db-502cd23a7eed.png)

**Comment by guppy42** [comment]
@coolkidsrule101 why the spam?

@MrReskill you need to post the actual out put a screenshot isn't helpfull at all, just dump the log on something like pastebin and put a link here.

**Comment by coolkidsrule101** [comment]
Ok good job

**Comment by coolkidsrule101** [comment]
And what spam

**Comment by MrReskill** [comment]
https://pastebin.com/Q5awh9RM

**Comment by guppy42** [comment]
your missing libfmod.so.9.6 it should be in;
/usr/local/share/mcpelauncher/libs/native/

that kind of wierd as e1569af specifically fixes this issue.

Are you using the "master" branch? and have you done "make install" ?

**Comment by MrReskill** [comment]
i use the master and i have been typed cmake . and make

**Comment by guppy42** [comment]
try 'sudo make install' it may be that you can no longer run it directly out of the source dir with out having it installed also.

**Comment by MrReskill** [comment]
Yet the make is well done!

**Comment by guppy42** [comment]
I honestly cannot figure you what your last message is supposed to mean - I hope it means it worked for you :)

**Comment by MrReskill** [comment]
No, I already did the make and it was ok

**Comment by MrReskill** [comment]
i tried reinstall

**Comment by guppy42** [comment]
ahh.

setup_bin_libs.sh downloads the file you are missing, so I'm guessing you deleted everything or did a fresh checkout. in this case you need to do all the steps again!

if you want fresh code to work with do;
```
  git clone --recursive https://github.com/MCMrARM/mcpelauncher-linux.git
```

if you have already checked the source code out - make sure it's up to date;
```
  git pull
  git submodule init
  git submodule update
```

No in the source dir "mcpelauncher" you need to do all the steps again

```
  ./setup_cef.sh
  ./download_icon.sh
  ./setup_bin_libs.sh
  cmake .
  make
```

At the end you can optionally choose to install the launcher so you do not have to navigate to the source directory every time by doing 

`sudo make install`

Have fun!

**Comment by MrReskill** [comment]
So that allows me to reinstall the right version?

**Comment by guppy42** [comment]
that will rebuild the latest version mcpelauncher ( and optionally install it )

It has no effect on the version of minecraft installed - to update it to the latest version add "setup" as a parameter when calling mcpelauncher ie; `./mcpelauncher setup` to launch from source or `mcpelauncher setup` to launch the installed version.

it is possible to have multiple version of minecraft on your system see #225 for how

**Comment by MrReskill** [comment]
how to upgrade it call to me is not the last version

**Comment by MrReskill** [comment]
![capture du 2017-12-21 22-53-09](https://user-images.githubusercontent.com/16138643/34276458-c02cdc7e-e6a1-11e7-8ffb-047c22b1a07a.png)

but when i try to update it say to me "this is the highters version"

**Comment by guppy42** [comment]
I've written how to upgrade both the launcher and minecraft above - if you don't understand it you may need to find somebody to translate it into French for you

**Comment by guppy42** [comment]
Let.me.kmow if need more help

**Comment by coolkidsrule101** [comment]
Do what I do save the worlds and delete everything that is mcpelauncher and
redo it again for mcpe 1.2.8

**Comment by coolkidsrule101** [comment]
Wait wait nooo don't do what I said your cmake is need to be 3.6.9 or
higher upgrade it please than try again

**Comment by MrReskill** [comment]
but when i want to update cmake he say "You're version of cmake is the highter"

**Comment by coolkidsrule101** [comment]
Uninstall cmake then install the cmake that is required

**Comment by guppy42** [comment]
Or follow the instructions for installing a local cmake in the wiki link I provided..

**Comment by MrReskill** [comment]
I get this
![capture du 2017-12-22 20-37-59](https://user-images.githubusercontent.com/16138643/34310473-0d893718-e758-11e7-8de1-9f370e890ab8.png)
 error

**Comment by guppy42** [comment]
@MrReskill please just post the error part as plain text, I don't understand french at all and having to type the entire mess into google translate manually makes me annoyed.

If google translate is accurate the problem you have is that you have put "libegl1-mesa-dev:i386" on hold this prevents it from being updated - I'm sure you had a good reason for it ( or maybe you just copied some random code off the net ), regardless you cannot proceed until you solve this issue.

**Comment by guppy42** [comment]
if you do `dpkg -l | grep ^h` it should show you a list of all packages on your system that are held
to "unhold" a package use `sudo apt-mark unhold packagename` (instead of packagename use it's actual name) do NOT do this unless you know there is no reason for package to be held, as if it's held for a reason something can break on your system

**Comment by MrReskill** [comment]
Okay, I must type sudo apt-mark unhold libegl1-mesa-dev:i386

**Comment by MrReskill** [comment]
?

**Comment by MrReskill** [comment]
When i type the commands he say:

libegl1-mesa-dev:i386 était déjà marqué comme non figé.

**Comment by guppy42** [comment]
yes if

- you wish no longer to hold that package
- it's actually being held ( what did the `dpkg -l | grep ^h` write if anything )

**Comment by MrReskill** [comment]
dpkg -l | grep ^h don't have write

**Comment by guppy42** [comment]
sems to mean it's not held - you should properly secure the help of somebody who speaks french / is french and knows Linux

**Comment by MrReskill** [comment]
I finally managed to install mcbe. Thanks for your help ! On the other hand, how do I create an icon to launch from the taskbar? Thank you so much !

**Comment by guppy42** [comment]
Create a launcher to /usr/local/bin/mcpelauncher there is an icon you can use in /use/local/share/pixmaps/

**Comment by coolkidsrule101** [comment]
Look at the compiling instructions grow the website
